articles/ros2_threat_model.md

```diff
@@ -0,0 +1,3291 @@
+    <td>Sensor data must not be accessed by attackers.</td>
+  </tr>
+  <tr>
+    <td>Robot Data Stores Privacy</td>
+    <td>Robot persistant data (logs, software, etc.) must not be accessible by
+        unauthorized actors.</td>
+  </tr>
+
+  <tr><th colspan="2">Integrity</th></tr>
+  <tr>
+    <td>Physical Safety</td>
+    <td>The robotic system must not harm its users or environment.</td>
+  </tr>
+  <tr>
+    <td>Robot Actuators Command Integrity</td>
+    <td>Unallowed actors should not be able to control the robot actuators.</td>
+  </tr>
+  <tr>
+    <td>Robot Behavior Integrity</td>
+    <td>The robotic system must not allow attackers to disrupt its tasks.</td>
+  </tr>
+  <tr>
+    <td>Robot Data Stores Integrity</td>
+    <td>No attacker should be able to alter robot data.</td>
+  </tr>
+
+  <tr><th colspan="2">Availability</th></tr>
+  <tr>
+    <td>Compute Capabilities</td>
+    <td>Robot embedded and distributed (e.g. cloud) compute resources.
+    Starving a robot from its compute resources can prevent it from operating
+    correctly.
+    </td>
+  </tr>
+  <tr>
+    <td>Robot Availability</td>
+    <td>The robotic system must answer commands in a reasonable time.</td>
+  </tr>
+  <tr>
+    <td>Sensor Availability</td>
+    <td>Sensor data must be available to allowed actors shortly after being
+    produced.</td>
+  </tr>
+</table>
+</div>
+
+#### Entry Points
+
+Entry points describe the system attack surface area (how do actors interact
+with the system?).
+
+<div class="table">
+<table class="table">
+  <thead>
+    <th>Name</th>
+    <th>Description</th>
+  </thead>
+  <tr>
+    <td>Robot Components Communication Channels</td>
+    <td>Robotic applications are generally composed of multiple components
+    talking over a shared bus. This bus may be accessible over the robot
+    WAN link.</td>
+  </tr>
+  <tr>
+    <td>Robot Administration Tools</td>
+    <td>Tools allowing local or remote users to connect to the robot
+    computers directly (e.g. SSH, VNC).</td>
+  </tr>
+  <tr>
+    <td>Remote Application Interface</td>
+    <td>Remote applications (cloud, smartphone application, etc.) can be
+    used to read robot data or send robot commands (e.g. cloud REST API,
+    desktop GUI, smartphone application).</td>
+  </tr>
+  <tr>
+    <td>Robot Code Deployment Infrastructure</td>
+    <td>Deployment infrastructure for binaries or configuration files are
+    granted read/write access to the robot computers filesystems.</td>
+  </tr>
+  <tr>
+    <td>Sensors</td>
+    <td>Sensors are capturing data which usually end up being injected into
+    the robot middleware communication channels.</td>
+  </tr>
+  <tr>
+    <td>Embedded Computer Physical Access</td>
+    <td>External (HDMI, USB...) and internal (PCI Express, SATA...) ports.</td>
+  </tr>
+</table>
+</div>
+
+### Robot Application Components and Trust Boundaries
+
+The system is divided into hardware (sensors, actuators), multiple components
```

Comment:
Shouldn't you add the general purpose-computer of line https://github.com/ros2/design/pull/218/files#diff-60aa2b3022e7728b236f31c650141eceR69  to the hardware here? 
```suggestion
The system is divided into hardware (embedded general-purpose computer, sensors, actuators), multiple components
```